docs/peer_forwarder.md

```diff
@@ -0,0 +1,121 @@
+
+## Peer Forwarder
+An HTTP service which performs peer forwarding of `Event` between Data Prepper nodes for aggregation. Currently, supported by `aggregate`, `service_map_stateful`, `otel_trace_raw` processors.
+
+Peer Forwarder groups events based on the identification keys provided the processors.
+For `service_map_stateful` and `otel_trace_raw` it's `traceId` by default and can not be configured.
+It's configurable for `aggregate` processor using `identification_keys` configuration option. You can find more information about identification keys [here](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/aggregate-processor#identification_keys).
+
+---
+
+Presently peer discovery is provided by either a static list or by a DNS record lookup or AWS Cloudmap.
+
+### Static discovery mode
+Static discover mode allows Data Prepper node to discover nodes using a list of IP addresses or domain names.
+```yaml
+peer_forwarder:
+  discovery_mode: static
+  static_endpoints: ["data-prepper1", "data-prepper2"]
+```
+
+### DNS lookup discovery mode
+DNS discovery is recommended when scaling out a Data Prepper cluster. The core concept is to configure a DNS provider to return a list of Data Prepper hosts when given a single domain name.
+This is a [DNS A Record](https://www.cloudflare.com/learning/dns/dns-records/dns-a-record/) which indicates a list list of IP addresses of a given domain.
+```yaml
+peer_forwarder:
+  discovery_mode: dns
+  domain_name: "data-prepper-cluster.my-domain.net"
+```
+
+### AWS Cloud Map discovery mode
+
+[AWS CloudMap](https://docs.aws.amazon.com/cloud-map/latest/dg/what-is-cloud-map.html) provides API-based service discovery as well as DNS-based service discovery.
+
+Peer forwarder can use the API-based service discovery. To support this you must have an existing
+namespace configured for API instance discovery. You can create a new one following the instructions
+provided by the [CloudMap documentation](https://docs.aws.amazon.com/cloud-map/latest/dg/working-with-namespaces.html).
+
+Your pipeline configuration needs to include:
```

Comment:
Should it be in data prepper server configuration instead?

Reply:
Good catch!